specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-10-17-preview/workbookTemplates_API.json

```diff
@@ -0,0 +1,467 @@
+            "in": "body",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/WorkbookTemplateUpdateParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The workbook template definition updated.",
+            "schema": {
+              "$ref": "#/definitions/WorkbookTemplate"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/WorkbookError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookTemplateUpdate": {
+            "$ref": "./examples/WorkbookTemplateUpdate.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "WorkbookTemplateResource": {
+      "properties": {
+        "resourceId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource name."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "WorkbookTemplatesListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/WorkbookTemplate"
+          },
+          "description": "An array of workbook templates."
+        }
+      },
+      "description": "WorkbookTemplate list result."
+    },
+    "WorkbookTemplate": {
+      "description": "An Application Insights workbook template definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/WorkbookTemplateResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a workbook template for an Azure resource.",
+          "$ref": "#/definitions/WorkbookTemplateProperties"
+        }
+      }
+    },
+    "WorkbookTemplateProperties": {
+      "description": "Properties that contain a workbook template.",
+      "required": [
+        "id",
+        "templateData",
+        "galleries"
+      ],
+      "properties": {
+        "id": {
```

Comment:
"id" should not be used inside the property bag as it can be confusing with the top level resource "id"

Reply:
That is why I named top level id resourceId. I can change this to templateId and the top level can be id?